### PR TITLE
internal/asm/...: add documentation

### LINKED ISSUE #47: internal/asm/...: document packages and functions
Currently the asm packages have zero documentation. There are four ways we can go with this:

1. Leave it as is.
2. Add a package-level stanza with an explanation of the absence of documentation and that users who are interested should look at the source.
3. Add package-level documentation and reflect the naive Go source for each function. The justification for this is that they functions are all trivial and they are intended as replacements for the Go.
4. Add package-level documentation and write English documentation.

My preferred options are 2 or 3.

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

This may seem a little odd, but it arose out of a discussion I was having with someone who wanted to contribute to the complex blas implementation. These comments will help people working on the blas and floats packages

Fixes #47.

### CODE REVIEW

File: asm/c128/doc.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package c128 provides complex128 vector primatives.
```
Comment (vladimir-ch) [inline asm/c128/doc.go]: primitives here and in the other packages

File: asm/f64/stubs_amd64.go [hunk 2]
```diff
@@ -6,42 +6,160 @@
...
+//  return sum
 func DotInc(x, y []float64, n, incX, incY, ix, iy uintptr) (sum float64)
 
+// L1Norm is
+//  var norm float64
+//  for i, v := range s {
+//  	norm += math.Abs(t[i] - v)
+//  }
+//  return norm
 func L1Norm(s, t []float64) float64
```
Comment (vladimir-ch) [inline asm/f64/stubs_amd64.go]: These two should be called L1Dist (or L1Distance) and LinfDist.

File: asm/f64/stubs_noasm.go [hunk 3]
```diff
@@ -8,32 +8,59 @@ package f64
 
 import "math"
 
+// AbsSum is
+//  for _, v := range x {
+//  	sum += math.Abs(v)
+//  }
+//  return sum
 func AbsSum(x []float64) (sum float64) {
```
Comment (vladimir-ch) [inline asm/f64/stubs_noasm.go]: This one could be called L1Norm.

**Review by vladimir-ch — APPROVED** [review]

File: asm/c128/doc.go (hunk 1 above)
Comment (kortschak) [reply asm/c128/doc.go]: Oops. Fixed.

File: asm/f64/stubs_amd64.go (hunk 2 above)
Comment (kortschak) [reply asm/f64/stubs_amd64.go]: Done.

File: asm/f64/stubs_noasm.go (hunk 3 above)
Comment (kortschak) [reply asm/f64/stubs_noasm.go]: Done.

**Comment by kortschak** [comment]
PTAL

**Review by Kunde21 — APPROVED** [review]
LGTM

**Pull request merged**